sensor_test.py

```diff
@@ -0,0 +1,208 @@
+    """
+    1. Set sensor name to "new_name".
+    2. Get sensor_info.
+    3. Validate that current sensor name matches the name set in Step 1.
+    """
+    print("Set  sensor name different from default")
+    new_sensor_name = "Updated_sensor_name"
+    set_sensor_name(new_sensor_name)
+
+    print("Get sensor_info")
+    updated_sensor_info = get_sensor_info()
+
+    print("Validate that current sensor name matches the name set in Step 1")
+    current_sensor_name = updated_sensor_info.get("name")
+    assert (
+        current_sensor_name == new_sensor_name
+    ), f"Sensor name was not updated, expected name:{new_sensor_name}"
+
+
+def test_set_sensor_reading_interval(
+    get_sensor_info, set_sensor_reading_interval, get_sensor_reading
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+    print("Set sensor reading interval to 1")
+    new_reading_interval = 1
+    # set_sensor_reading_interval(new_reading_interval)
+    #
+    print("Get sensor info")
+    # updated_sensor_info = get_sensor_info()
+    updated_sensor_info = set_sensor_reading_interval(new_reading_interval)
+
+    print("Validate that sensor reading interval is set to interval from Step 1")
+    current_reading_interval = updated_sensor_info.get("reading_interval")
+    assert (
+        current_reading_interval == new_reading_interval
+    ), f"Sensor reading interval was not updated, expected interval {new_reading_interval}"
+
+    print("Get sensor reading")
+    sensor_reading_before_wait = get_sensor_reading()
+
+    print("Wait for interval specified in Step 1")
+    sleep(new_reading_interval)
+
+    print("Get sensor reading")
+    sensor_reading_after_wait = get_sensor_reading()
+
+    print("Validate that reading from tep 4 doesn't equal reading from Step 6")
+    assert (
+        sensor_reading_before_wait != sensor_reading_after_wait
+    ), f"Sensor readings after {new_reading_interval} second of waiting are equal to readings before waiting"
+
+
+def test_update_sensor_firmware(get_sensor_info, update_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+    print("Get original sensor firmware version")
```

Comment:
See if you can simplify this even further to the following:
1. Make a `while` loop request firmware updates up until sensor version is 15.
2. After the loop is done, request another firmware update and validate that: (a) sensor says that it's already at max version; (b) sensor version stays at max value.